diffusion/datasets/image_caption.py

```diff
@@ -43,6 +43,8 @@ class StreamingImageCaptionDataset(StreamingDataset):
         image_key (str): Key associated with the image in the streaming dataset. Default: ``'image'``.
         caption_key (str): Key associated with the caption in the streaming dataset. Default: ``'caption'``.
         sdxl (bool): Whether or not we're training SDXL. Default: `False`.
+        use_e5 (bool): Whether to use e5-large-v2 tokenizer instead of openai CLIP.
```

Comment:
Having this is a boolean has bad code smell, is there anyway we can automatically enable  this boolean based on the sub_tokenizer? or by adding a substring to the end of the tokenizer maybe?